### PR TITLE
[Example 2]: Multiple command interfaces using prepare - perform switch

### PR DESCRIPTION
**Pull request by mahaarbo**
Uses mode switching based on [ros2_control/#348](https://github.com/ros-controls/ros2_control/pull/348).

### CODE REVIEW

File: ros2_control_demo_robot/config/rrbot_multi_interface_forward_controllers.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      type: forward_command_controller/ForwardCommandController
+
+    forward_command_controller_illegal1:
+      type: forward_command_controller/ForwardCommandController
+
+    forward_command_controller_illegal2:
+      type: forward_command_controller/ForwardCommandController
+
+    joint_state_controller:
+      type: joint_state_controller/JointStateController
```
Comment (bmagyar) [inline ros2_control_demo_robot/config/rrbot_multi_interface_forward_controllers.yaml:21]:
```suggestion
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
```

**Review by bmagyar — APPROVED** [review]
I've tested this this morning, added some basic docs, looks good to me!

**Comment by destogl** [comment]
I just rebased this to master and applied format. There is still an issue with `ament_cppcheck` which I don't understand why... (@bmagyar, @Karsten1987 : could you please take a look?)

The error is the following: `[ros2_control_demo_hardware/src/rrbot_system_multi_interface.cpp:162]: (error: mismatchingContainers) Iterators of different containers are used together.`

**Review by destogl — APPROVED** [review]
Looks good!

**Comment by bmagyar** [comment]
Thanks for everyone who contributed!

**Comment by mahaarbo** [comment]
Thanks for the help!

**Pull request merged**